From 181fa663ac6d803a4ddf4c79c3258a3297612b03 Mon Sep 17 00:00:00 2001
From: Frank Binns <frank.binns@imgtec.com>
Date: Fri, 28 Aug 2026 08:27:47 +0100
Subject: [PATCH] zink: enable renderpass tracking for PVR

This is already enabled for the proprietary Imagination Vulkan driver, so do the
same for the Mesa Vulkan driver, as it brings performance benefits.

Signed-off-by: Frank Binns <frank.binns@imgtec.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/44071>
(cherry picked from commit 90320e129e94e2dd68868db068c7625f4054e547)
Signed-off-by: Icenowy Zheng <zhengxingda@iscas.ac.cn>
---
 src/gallium/drivers/zink/zink_screen.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/gallium/drivers/zink/zink_screen.c b/src/gallium/drivers/zink/zink_screen.c
index 1f308e5b123..7399525e422 100644
--- a/src/gallium/drivers/zink/zink_screen.c
+++ b/src/gallium/drivers/zink/zink_screen.c
@@ -3093,6 +3093,7 @@ init_driver_workarounds(struct zink_screen *screen)
    case VK_DRIVER_ID_MESA_TURNIP:
    case VK_DRIVER_ID_MESA_PANVK:
    case VK_DRIVER_ID_MESA_V3DV:
+   case VK_DRIVER_ID_IMAGINATION_OPEN_SOURCE_MESA:
    case VK_DRIVER_ID_IMAGINATION_PROPRIETARY:
    case VK_DRIVER_ID_QUALCOMM_PROPRIETARY:
    case VK_DRIVER_ID_BROADCOM_PROPRIETARY:
-- 
2.56.0

